src/starwars/pages/SwCharacters.tsx

```diff
@@ -0,0 +1,60 @@
+import { Tab, Tabs } from "react-bootstrap";
+import { Fragment, useEffect, useState } from "react";
+import Character from "../models/Character"
+import { getById } from "../services/SWService";
+import characters from "../data/CharacterDictionary";
+import CharacterDetails from "../components/characterDetails/CharacterDetails";
+import LoadingSpinner from "../components/loadingSpinner/LoadingSpinner";
+import { readLocalStorage, writeLocalStorage } from "../../common/services/LocalStorageService";
+import { useParams, useNavigate } from "react-router-dom";
+
+export default function SwCharacters() {
+  const navigate = useNavigate();
+  const { tabKey } = useParams<{ tabKey: string }>();
+  const [key, setKey] = useState(tabKey ?? readLocalStorage('lastTab') ?? 'luke');
+  const [isLoading, setIsLoading] = useState(false);
+  const [character, setCharacter] = useState<Character>();
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      const sendTime = Date.now();
```

Comment:
mby encapsulate this logic in custom hook that returns data, isLoading and potential error. call is sth like useDeplayedCall(() => getById('people', characters.get(key));